source/solvers/fluid_dynamics_matrix_based.cc

```diff
@@ -103,10 +109,36 @@ FluidDynamicsMatrixBased<dim>::setup_dofs_fd()
                       this->mpi_communicator);
     }
 
+  // Reinitialize vectors used for the SDIRK methods
+  this->sdirk_vectors.sum_bi_ki.reinit(this->locally_owned_dofs,
+                                       this->locally_relevant_dofs,
+                                       this->mpi_communicator);
+  this->sdirk_vectors.temp_sum_bi_ki.reinit(this->locally_owned_dofs,
+                                            this->mpi_communicator);
+
+
+  this->sdirk_vectors.sum_over_previous_stages.reinit(
+    this->locally_owned_dofs,
+    this->locally_relevant_dofs,
+    this->mpi_communicator);
+  this->sdirk_vectors.temp_sum_over_previous_stages.reinit(
+    this->locally_owned_dofs, this->mpi_communicator);
+
+  this->sdirk_vectors.locally_owned_for_calculus.reinit(
+    this->locally_owned_dofs, this->mpi_communicator);
```

Comment:
I would reinitialize these vector only if the method is sdirk. you have this nice function in your simulation control to check it :)
Will save calculations for the rest of simulations.